supply/lib/supply/client.rb

```diff
@@ -5,7 +5,67 @@
 require 'net/http'
 
 module Supply
-  class Client
+  UI = FastlaneCore::UI
+
+  class GoogleServiceClient
+    def self.make_from_config(params: nil)
+      unless params[:json_key] || params[:json_key_data]
+        UI.important("To not be asked about this value, you can specify it using 'json_key'")
+        params[:json_key] = UI.input("The service account json file used to authenticate with Google: ")
+      end
+
+      if params[:json_key]
+        service_account_json = File.open(File.expand_path(params[:json_key]))
+      elsif params[:json_key_data]
+        service_account_json = StringIO.new(params[:json_key_data])
+      end
+
+      return self.new(service_account_json: service_account_json, params: params)
+    end
+
+    # Initializes the service and its auth_client using the specified information
+    # @param service_account_json: The raw service account Json data
+    def initialize(service_account_json: nil, params: nil)
```

Comment:
(see comment below this first)

It might also be nice to add `scope` and `service` as a parameter to this `initialize` method so that subclasses know that they need to be passed instead/set.

Reply:
I had problems with subclasses having different parameters on methods - but that was probably me not knowing how ruby works. (I guess I messed up the method param signatures or something, `params` vs. `params:nil` etc)